manual/root_macros_and_shared_libraries/index.md

```diff
@@ -40,41 +40,39 @@ The name of the ROOT macro and the file name (without file extension) in which t
 
 ### Executing ROOT macros
 
-You can execute a ROOT macro:
+You can execute a ROOT macro in one of three ways:
 
-  - at the system prompt,
-  - at the ROOT prompt,
-  - by loading it to a ROOT session.
 
-To execute a ROOT macro at the system prompt, type:
+1. Execute a ROOT macro at the system prompt:
 
    ```
-   root MacroName.C
+   $ root MacroName.C
    ```
 
--- or --
-
-To execute a ROOT macro at the ROOT prompt, type:
+2. Execute a macro at the ROOT prompt:
 
    ```
-   .x MacroName.C
+   root [0] .x MacroName.C
    ```
 
--- or --
-
-To load a ROOT macro to a ROOT session, type (at the ROOT prompt):
+3. Load a macro from within a ROOT session and then call the function:
 
    ```
-   .L MacroName.C
-   MacroName()
+   root [0] .L MacroName.C
+   root [1] MacroName()
    ```
 
-> **Note**
->
-> You can load multiple ROOT macros, as each ROOT macro has a unique name in the ROOT name space.
+   > **Note**
+   >
+   > You can load multiple ROOT macros, as each ROOT macro has a unique name in the ROOT namespace.
+
+It is also possible to pass parameters directly to the macro function:
+```
+$ root 'MacroName.C("some String", 12)'
+```
+
+In addition, you can execute a ROOT macro from a ROOT macro.
 
-In addition, you can:
- - [execute a ROOT macro from a ROOT macro](#executing-a-ROOT-macro-from-a ROOT-macro)
  - [execute a ROOT macro from the invocation of ROOT](#executing-a-root-macro-from-the-invocation-of-root)
 
 <p><a name="executing-a-ROOT-macro-from-a ROOT-macro"></a></p>
```

Comment:
I would change here: 

**Executing a ROOT macro from a ROOT macro**

You can execute a ROOT macro conditionally inside another ROOT macro by calling directly the interpreter using 
[TROOT::ProcessLine()](https://root.cern/doc/master/classTROOT.html#a32fc66033a13d1415e0ad523994dd0e5){:target="_blank"}.

`ProcessLine()` takes in addition to the code to be executed an optional parameter, which is a pointer to an `int` or to a
`TInterpreter::EErrorCode` to let you access the interpreter error code after an attempt to interpret.
It returns the result of the command casted to a `LongPtr_t`


Then in the example , would it not be better to add a case where we return something ?

Reply:
Ok good point, I have changed the example to pretend that `cernbuild` returns some error code that we will store in a new variable.